modules/registrars/openprovider/OpenProvider/API/ApiV1.php

```diff
@@ -121,14 +123,67 @@ public function call(string $cmd, array $args = []): ResponseInterface
             return $response;
         }
 
+
+        if (isset($reply['warnings'])) {
+            $this->add_warning($reply['warnings'], $cmd, $args);
+        }
+
         $data = $this->serializer->normalize($reply->getData());
         $response = $this->successResponse($response, $data);
-
         $this->log($cmd, $args, $response);
 
         return $response;
     }
 
+    /**
+     * @param array $warnings
+     * @param string $cmd
+     */
+    private function add_warning($warnings, $cmd, $args)
+    {
+        $title = "Warning: ";
+        $description = "";
+        $index = 1;
+        foreach ($warnings as $warn) {
+            if ($warn['code'] != 0) {
+                $description .= "Warning " . $index . ":" . "\nCode: " . $warn["code"] . "\nDescription: " . $warn["desc"]  . "\nData: " . $warn["data"] . "\n";
+            }
+        }
+        $cmd = preg_replace('/(?<!\s)([A-Z])/', ' $1', $cmd); // Insert space before capital letters
+        $cmd = ucwords($cmd); // Capitalize words
+        $title = $title . $cmd . " Command.";
+        if (isset($args['domain']['name']) && isset($args['domain']['extension'])) {
+            $domainName = $args['domain']['name'];
```

Comment:
better like
`if (isset($args['domain']) {
    $domainName = $args['domain']['name'] ?? null;
    $extension = $args['domain']['extension'] ?? null;
}`

Also, mostly there should be function to concat $domainName and $extension. but not sure

Reply:
This change has been added